src/Core/Services/Implementations/MultiServiceMailDeliveryService.cs

```diff
@@ -0,0 +1,53 @@
+﻿using System;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Bit.Core.Settings;
+using Microsoft.AspNetCore.Hosting;
+using System.Net.Http;
+using Bit.Core.Models.Mail;
+using Bit.Core.Utilities;
+
+namespace Bit.Core.Services
+{
+    public class MultiServiceMailDeliveryService : IMailDeliveryService
+    {
+        private readonly IMailDeliveryService _sesService;
+        private readonly IMailDeliveryService _postalService;
+        private readonly int _postalPercentage;
+
+        private static Random _random = new Random();
+
+        public MultiServiceMailDeliveryService(
+            GlobalSettings globalSettings,
+            IWebHostEnvironment hostingEnvironment,
+            IHttpClientFactory httpClientFactory,
+            ILogger<AmazonSesMailDeliveryService> sesLogger,
+            ILogger<PostalMailDeliveryService> postalLogger)
+        {
+            _sesService = new AmazonSesMailDeliveryService(globalSettings, hostingEnvironment, sesLogger);
+
+            if (CoreHelpers.SettingHasValue(globalSettings.Mail?.PostalApiKey) &&
```

Comment:
This check seems redundant as this implementation isn't added to the DI container unless this value is configured during startup (similar to the assumption above that the SES provider is configured already by the same service collection extension method). Would there be a time when this provider is injected w/o SES and Postal both configured?

Reply:
There wouldn't. Originally I was thinking maybe you wanted to turn off postal but still use `MultiServiceMailDeliveryService`, but that can still be done outsidfe of this scope.

Reply:
removed